File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    dict = {}
    for name, id in clothes:
        if id not in dict:
            dict[id] =[name]
        else:
            dict[id].append(name)
    

    for v in dict.values():
        answer = (len(v)+1)*answer
    
    
    
    return answer-1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        
        HashMap<String,Integer> map = new HashMap<>();
        int n = clothes.length;
        for(int i=0; i<n ;i++){
            map.put(clothes[i][1], map.getOrDefault(clothes[i][1],0)+1);
        }
        for(String key: map.keySet()){
            answer *= (map.get(key)+1);
        }
        
        answer-=1;
        
        return answer;
    }
}